bin/main.rb

```diff
@@ -1,3 +1,58 @@
 #!/usr/bin/env ruby
 
-puts 'Hello World'
+board = [[1, 4, 7], [2, 5, 8], [3, 6, 9]]
+
+def display_grid(board)
+  board.each do |n|
+    n.each do |j|
+      print "#{j} "
+    end
+    print "\n"
+  end
+end
+
+game_on = true
+colors = %w[X O]
+puts 'Welcome to tic-tac-toe game'
+display_grid(board)
+puts "\nEnter Player-1 name:"
+player1 = gets.chomp
+puts "\nPlease select X or O"
+color1 = gets.chomp
+puts "\n#{player1}, your color is #{color1}"
+colors.delete(color1)
+puts "\nEnter Player-2 name:"
+player2 = gets.chomp
+color2 = colors[0]
+puts "\n#{player2}, your color is #{color2}"
+
+winner1 = false
+winner2 = false
+positions = { '1' => [0, 0], '2' => [1, 0], '3' => [2, 0],
+              '4' => [0, 1], '5' => [1, 1], '6' => [2, 1],
+              '7' => [0, 2], '8' => [1, 2], '9' => [2, 2] }
+
+while game_on
+  9.times do
```

Comment:
:warning: Using ```ruby 9.times do ``` is making the functionality dependent on specific criteria. We want our functionalities fully flexible. To do so the best and optimal way is to use terminal variables. So If you are planning to use the loop you have created in your upcoming milestone, that will create a mess.[But I am assuming that you are using it in this milestone only to mimic the view. If you need more info, don't hesitate to knock me].